File: Weekly_Contests/Weekly_327/Q2.py
# Written by: Jian-Yuan Lin
# Date: 2023.01.07

class Solution:
    def maxKelements(self, nums: List[int], k: int) -> int:
        maxheap = []
        for num in nums:
            heapq.heappush(maxheap, -num)
        
        ans = 0
        for _ in range(k):
            val = -heapq.heappop(maxheap)
            ans += val
            heapq.heappush(maxheap, -ceil(val/3))
        return ans

File: Weekly_Contests/Biweekly_99/Q2.cpp
class Solution {
public:
    long long coloredCells(int n) {
        vector<long long> dp(n+1, 0);
        dp[1] = 1;
        for (int i= 2; i <= n; i++){
            dp[i] = dp[i-1] + 4 * (i-1);
        }
        return dp[n];
    }
};